fix(aio): upgrade - remove broken hybrid router example

filipesilva opened the pull request:
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Other... Please describe: AIO ngUpgrade fix
```

**What is the current behavior?** (You can also link to an open issue here)
The current ngUpgrade guide setup has issues with zones for dual router apps.


**What is the new behavior?**
Dual router advice in ngUpgrade guide is removed, and bootstrap instructions are changed.


**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...


**Other information**:
This is a ported PR from the angular/angular.io repo https://github.com/angular/angular.io/pull/3541, augmented with example removal.

/cc @wardbell @petebacondarwin @Foxandxss

wardbell commented:
@petebacondarwin should review the code changes (and some prose) to confirm that this is what we should be saying.

I noticed a few small prose things I'd like to fix. The most important, though, is that it hasn't gone through the "we" -> "you" sweep. I'll take care of that in an attached commit.

petebacondarwin commented on aio/content/examples/upgrade-module/src/app/divide-routes/app.module.ts:
```diff
@@ -31,10 +31,17 @@ class HybridUrlHandlingStrategy implements UrlHandlingStrategy {
+  declarations: [ AppComponent ]
 })
-export class AppModule { }
+export class AppModule extends UpgradeModule {
```
No need for the extends here.

filipesilva replied:
Fixed

petebacondarwin commented on aio/content/examples/upgrade-module/src/app/divide-routes/app.module.ts:
```diff
@@ -31,10 +31,17 @@ class HybridUrlHandlingStrategy implements UrlHandlingStrategy {
-export class AppModule { }
+export class AppModule extends UpgradeModule {
+  constructor(injector: Injector, zone: NgZone) {
+    super(injector, zone);
```
no need for the super call here (once we remove the extends above.

filipesilva replied:
Fixed

petebacondarwin commented on aio/content/examples/upgrade-module/src/app/upgrade-static/hero-detail.component.ts:
```diff
@@ -21,5 +22,15 @@ export class HeroDetailDirective extends UpgradeComponent {
+  ngOnInit() { super.ngOnInit(); }
+  ngOnChanges(changes: SimpleChanges) { super.ngOnChanges(changes); }
+  ngDoCheck() { super.ngDoCheck(); }
+  ngOnDestroy() { super.ngOnDestroy(); }
```
I think that the latest Angular will actually cope without us manually providing these. @gkalpak  is that correct?

gkalpak replied:
Correct.

petebacondarwin commented on aio/content/guide/upgrade.md:
```diff
@@ -522,10 +521,16 @@ will result in the same thing:
-Now introduce Angular to the project. Inspired by instructions in
-[the Setup](guide/setup), you can selectively copy in material from the
+To begin converting our AngularJS application to a hybrid, we need to load the Angular framework.
+You can see how this can be done with SystemJS by following the instructions in [the Setup](setup.html),
```
the link should be to `guide/setup`.

petebacondarwin commented on aio/content/guide/upgrade.md:
```diff
@@ -1158,83 +1161,16 @@ Module Factory:
-
-The AoT metadata collector will not detect lifecycle hook methods on a parent class' prototype,
-so in order for upgraded components to work we needs to implement the lifecycle hooks
+Finally, the AoT metadata collector does not detect lifecycle hook methods on the prototype of a parent class.
```
We need to check whether this is still true. I am sure that @gkalpak told me it was no longer needed.

petebacondarwin commented on aio/content/guide/upgrade.md:
```diff
@@ -2244,47 +2173,33 @@ which recommends that you create a `NgModule` dedicated to router configuration
-And since `PhoneListComponent` isn't being rendered from a `<phone-list>` tag anymore,
-but rather routed to, we can do away with it's Angular selector as well.
+We must also remove the bootstrap of the AngularJS module from `ngDoBootstrap()` in `app.module.ts`
+and the `UpgradeModule` import.
```
I am not sure that these two sentences are correct.

petebacondarwin replied:
Hmm, I see that what we are really describing now is a non-ngUpgrade application. But that doesn't really match the headings above.

petebacondarwin reviewed (changes requested):
Thanks for taking this on @filipesilva !

The bit where we have removed the dual-router still doesn't quite read correctly IMO.
There are a few other nitpicks and the PR needs rebasing and conflicts resolving.

gkalpak commented on aio/content/examples/upgrade-module/src/app/upgrade-static/hero-detail.component.ts:
```diff
@@ -21,5 +22,15 @@ export class HeroDetailDirective extends UpgradeComponent {
+  // #docregion hero-detail-upgrade
 }
-// #enddocregion hero-detail-upgrade
+// #enddocregion hero-detail-upgrade		  // #enddocregion hero-detail-upgrade
```
Too much `enddocregion hero-detail-upgrade`?

wardbell commented:
@petebacondarwin As I write, `yarn docs` hangs when run with this PR.

filipesilva commented:
@petebacondarwin heya, I added the code changes you requested, but unsure of the problems in the prose. Can you give me a bit of direction on that?

gkalpak commented:
@filipesilva, (as discussed with @petebacondarwin) I will go over this tonight and provide feedack/suggest changes.

gkalpak commented:
I have rebased on master and pushed a commit with several minor fixes. Other than that LGTM (assuming the code works - I haven't run any code 😇).

gkalpak committed:
docs(aio): rework of the upgrade guide

This commit was worked on by a number of people including
@filipesilva, @gkalpak and @wardbell. It contains changes that:

* remove unused files,
* fix the bootstrap approach to ensure that bootstrap is in the correct Zone
* fix unclosed code-example tags
* replace use of "we" with "you"
* remove broken dual router example

Related to angular/angular.io#3541

mary-poppins commented:
The angular.io preview for 67c1c350dd650d7724831a03eefb507a032a36ad is available [here][1].

[1]: https://pr16409-67c1c350dd650d7724831a03eefb507a032a36ad.ngbuilds.io/

mary-poppins commented:
The angular.io preview for 426aec3a1762122cc1d99f423b988e1f29ebe58d is available [here][1].

[1]: https://pr16409-426aec3a1762122cc1d99f423b988e1f29ebe58d.ngbuilds.io/

gkalpak commented:
:tada:

filipesilva commented:
Awesomesauce, great work everyone 👍
